### PR TITLE
Update stylecop version and minor fixes

### PR DESCRIPTION
**Pull request by tamirkamara**
The following changes are proposed:

* Stylecop version to support newer versions of c#
* Logging params in Pascel Case
* Using var rather than explicit types
* More specific exceptions
* Remove unnecessary using statements
* Fix issues with comments
* Make methods as static if possible

### CODE REVIEW

**Comment by AsafMah** [comment]
I think this should wait until the top-hits PR is merged

**Comment by tamirkamara** [comment]
Just seen there're many files there too... we're going to wait.

**Review by eladiw — APPROVED** [review]

**Pull request closed without merging**
